karapace/kafka_rest_apis/__init__.py

```diff
@@ -292,7 +291,7 @@ def cluster_metadata(self, topics: Optional[List[str]] = None) -> dict:
                         content_type="application/json",
                         status=HTTPStatus.INTERNAL_SERVER_ERROR,
                     )
-        return copy.deepcopy(self._cluster_metadata)
+        return deepcopy(self._cluster_metadata)
```

Comment:
Does this need deepcopy at all? For me looks like `self._cluster_metadata` is assigned in this method with new value, and then returned.

Reply:
Cluster metadata gets mutated, so copy is needed. For example here: https://github.com/aiven/karapace/blob/2578805875d115146675d79c50df9501c3cc5edd/karapace/kafka_rest_apis/__init__.py#L676

I think this the copy is ok here. Alternative would be to create the copies where mutation occurs.

Reply:
Seems we're back here with #366

Reply:
I didn't review the code to check how the dict is used, is a shallow copy sufficient?